core/cache/machine.go

```diff
@@ -125,37 +124,45 @@ func (m *Machine) WatchApplicationLXDProfiles() (*MachineAppLXDProfileWatcher, e
 	if err != nil {
 		return nil, errors.Annotatef(err, "failed to get units to start MachineAppLXDProfileWatcher")
 	}
-	m.model.mu.Lock()
+
```

Comment:
I'm wondering if we'll miss data in between the above collection and the time when the subscriptions are done. Or if other model data we do get is changed under?

Using the machine mutex here... given the data being used, what is blocked from being changed?

Reply:
The pattern should be some form of setting up the subscriptions, then establishing the state of the world, rather than locking the world (model) until we are set up.

At the watcher instantiation, we access the `details` member, which is what we're protecting - the lock could be acquired later.